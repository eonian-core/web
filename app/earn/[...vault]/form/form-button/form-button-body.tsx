import type { ButtonProps } from '@nextui-org/react'
import { Button } from '@nextui-org/react'
import styles from './form-button.module.scss'

/** Minimal logic, will be used in skeleton */
export function FormButtonBody(props: ButtonProps) {
  return (
    <div className={styles.wrapper}>
      <Button
        auto
        color="primary"
        size="lg"
        className={styles.button}
        {...props}
        />
  </div>
  )
}
