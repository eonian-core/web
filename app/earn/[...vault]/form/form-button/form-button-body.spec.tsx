import React from 'react'
import { render } from '@testing-library/react'
import { FormButtonBody } from './form-button-body'

describe('FormButtonBody', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<FormButtonBody />)
    expect(asFragment()).toMatchSnapshot()
  })
})
