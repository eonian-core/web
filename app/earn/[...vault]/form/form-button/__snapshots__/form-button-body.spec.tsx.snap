// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormButtonBody renders correctly 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <button
      class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-unit-6 min-w-unit-24 h-unit-12 text-medium gap-unit-3 rounded-large [&>svg]:max-w-[theme(spacing.unit-8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none bg-primary text-primary-foreground data-[hover=true]:opacity-hover button"
      type="button"
    />
  </div>
</DocumentFragment>
`;
